anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/source/data/graphite/GraphiteSource.java

```diff
@@ -38,22 +39,37 @@
     private GraphiteClient graphiteClient;
 
     @Override
-    public List<DataSourceResult> getMetricData(String from, Integer maxDataPoints, String metric) {
-        List<GraphiteResult> graphiteResults = graphiteClient.getData(from, maxDataPoints, metric);
+    public List<DataSourceResult> getMetricData(int earliestTimeInSecs, int binSizeInSecs, String metric) {
+        int maxDataPoints = getMaxDataPointsPerDay(binSizeInSecs);
         List<DataSourceResult> results = new ArrayList<>();
-        if (graphiteResults.size() > 0) {
-            String[][] dataPoints = graphiteResults.get(0).getDatapoints();
-            //TODO Convert this to use JAVA stream
-            for (String[] dataPoint : dataPoints) {
-                Double value = MISSING_VALUE;
-                if (dataPoint[0] != null) {
-                    value = Double.parseDouble(dataPoint[0]);
+        for (int i = 0; i < earliestTimeInSecs; i += C.SECONDS_PER_DAY) {
+            List<GraphiteResult> graphiteResults = getDataFromGraphite(i, earliestTimeInSecs, maxDataPoints, metric);
+            if (graphiteResults.size() > 0) {
+                String[][] dataPoints = graphiteResults.get(0).getDatapoints();
+                //TODO Convert this to use JAVA stream
+                for (String[] dataPoint : dataPoints) {
+                    Double value = MISSING_VALUE;
+                    if (dataPoint[0] != null) {
+                        value = Double.parseDouble(dataPoint[0]);
+                    }
+                    long epochSeconds = Long.parseLong(dataPoint[1]);
+                    DataSourceResult result = new DataSourceResult(value, epochSeconds);
+                    results.add(result);
                 }
-                long epochSeconds = Long.parseLong(dataPoint[1]);
-                DataSourceResult result = new DataSourceResult(value, epochSeconds);
-                results.add(result);
             }
         }
         return results;
     }
-}
+
+    private List<GraphiteResult> getDataFromGraphite(int counter, int earliestTimeInSecs, int maxDataPoints, String metric) {
+        int from = earliestTimeInSecs - counter;
```

Comment:
As discussed offline, we should be using metric's timestamp as the final ABSOLUTE timestamp and counting backwards from that.